### ISSUE TITLE
Bug: [prefer-const] inconsistent `prefer-const` rule report

### ISSUE THREAD

**Issue by mikededo** [typescript-eslint/typescript-eslint#10426]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play#ts=5.6.2&fileType=.tsx&code=PQKgpgHgDg9gTgFzAEwAQEMTFQGzAjVAXlQCYBuIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQA7UQDNFoBaAYwHsA7ZAF3ShOkuknDAF8ROg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
// Using plain eslint, the following is not reported as an error

/* eslint/prefer-const: "error" */
/* exported a */ let a = 2;

// However, using this package, the same expression is reported as an error
/* exported a */ let a = 2; // 'a' is never reassigned. Use 'const' instead
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "prefer-const": "error",
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

In `eslint-plugin-svelte`, I am rewriting the `prefer-const` rule because, in Svelte, there are variable declarations that some users may prefer not to be reported as `const`. Since the `svelte-eslint-plugin` uses `@typescript-eslint`, I expect the base rule to exhibit the same behavior.

### Actual Result

The rule report is inconsistent.

### Additional Info

I've been researching for a bit and seems like `typescript-eslint` does not mark a `Variable` as `eslintUsed` the same way it does `eslint`, therefore comments such as `/* exported ` do not set `eslintUsed` to `true`.

**Comment by bradzacher** [comment]
[eslint playground](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IHByZWZlci1jb25zdDogZXJyb3IgKi9cblxuLyogZXhwb3J0ZWQgYSAqLyBsZXQgYSA9IDI7ICIsIm9wdGlvbnMiOnsicnVsZXMiOnt9LCJsYW5ndWFnZU9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoibW9kdWxlIiwicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnsianN4Ijp0cnVlfX19fX0=)

Looks like eslint core reports in exactly the same way.

**Comment by mikededo** [comment]
True, now I see that it only works as expected when source type is set to `script`. 

All good then.
